src/blop/dofs.py

```diff
@@ -228,15 +257,14 @@ def domain(self) -> Union[tuple[float, float], set[int], set[str], set[bool]]:
             else:
                 return TRANSFORM_DOMAINS[self.transform]
         else:
-            return self.trust_domain or self.search_domain
+            return self._domain
 
     def _transform(self, x: torch.Tensor, normalize: bool = True) -> torch.Tensor:
         if self.type != "continuous":
             raise ValueError("Cannot transform non-continuous DOFs.")
 
         # Since the DOF is continuous, we can safely assume the domains are tuples
-        domain = cast(tuple[float, float], self.domain)
-        x = torch.where((x > domain[0]) & (x < domain[1]), x, torch.nan)
+        x = torch.where(element_of(x, self.trust_domain), x, torch.nan)
```

Comment:
If you could add the type casting within `element_of` that would be great. Otherwise, you are undoing work done in #87